kernel/src/process.rs

```diff
@@ -84,6 +69,130 @@ pub unsafe fn load_processes(
     }
 }
 
+/// This trait is implemented by process structs.
+pub trait ProcessType {
+    /// Queue a callback for the process. This will be added to a per-process
+    /// buffer and passed to the process by the scheduler.
+    fn schedule(&self, callback: FunctionCall) -> bool;
+
+    /// Queue an IPC operation for this process.
+    fn schedule_ipc(&self, from: AppId, cb_type: IPCType);
+
+    /// Remove the scheduled operation from the front of the queue.
+    fn unschedule(&self) -> Option<Task>;
```

Comment:
What would you ever use unschedule for? (Why have this API?)

Reply:
Oh! I see where this is used, I'd assumed this was literally unscheduling the task, not grabbing it to be called. I might bikeshed this to `pop` or something akin to that from traditional queue interfaces that imply the result is intended to be used.

Reply:
It used to be called `dequeue_task`. But then you `schedule()` a `Callback` and then get it back by dequeuing a task, so `unschedule` was my attempt to make it more clear that it is the mirror to the `schedule` interface.

Reply:
how about switching to `enqueue`/`dequeue` instead? I agree with @ppannuto that `unschedule` is unintuitive and with @bradjc that `dequeue_task` doesn't match `schedule`